doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -724,58 +665,29 @@ the conversion in hypotheses :n:`{+ @ident}`.
          pattern (0 = 0).
          fold not.
 
-   .. tacv:: fold {+ @term}
+.. tacn:: pattern {+, @pattern_occs } {? @occurrences }
 
-      Equivalent to :n:`fold @term ; ... ; fold @term`.
+   Performs :math:`\beta`-expansion
+   (the inverse of :math:`\beta`-reduction) of the current goal (say :g:`T`)
+   for a set of :n:`@one_term`\s in the :n:`@pattern_occs`\s by:
 
-.. tacn:: pattern @term
-   :name: pattern
+   + replacing all occurrences of the :n:`@one_term`\s in :g:`T` with fresh variables
+   + abstracting these variables
+   + applying the abstracted goal to the :n:`@one_term`\s
 
-   This command applies to any goal. The argument :n:`@term` must be a free
-   subterm of the current goal. The command pattern performs :math:`\beta`-expansion
-   (the inverse of :math:`\beta`-reduction) of the current goal (say :g:`T`) by
-
-   + replacing all occurrences of :n:`@term` in :g:`T` with a fresh variable
-   + abstracting this variable
-   + applying the abstracted goal to :n:`@term`
-
-   For instance, if the current goal :g:`T` is expressible as
-   :math:`\varphi`:g:`(t)` where the notation captures all the instances of :g:`t`
-   in :math:`\varphi`:g:`(t)`, then :n:`pattern t` transforms it into
-   :g:`(fun x:A =>` :math:`\varphi`:g:`(x)) t`. This tactic can be used, for
-   instance, when the tactic ``apply`` fails on matching.
-
-.. tacv:: pattern @term at {+ @natural}
-
-   Only the occurrences :n:`{+ @natural}` of :n:`@term` are considered for
-   :math:`\beta`-expansion. Occurrences are located from left to right.
-
-.. tacv:: pattern @term at - {+ @natural}
-
-   All occurrences except the occurrences of indexes :n:`{+ @natural }`
-   of :n:`@term` are considered for :math:`\beta`-expansion. Occurrences are located from
-   left to right.
-
-.. tacv:: pattern {+, @term}
+   The :n:`@one_term`\s must be free subterms of the current goal.
 
    Starting from a goal :math:`\varphi`:g:`(t`:sub:`1` :g:`... t`:sub:`m`:g:`)`,
-   the tactic :n:`pattern t`:sub:`1`:n:`, ..., t`:sub:`m` generates the
+   :n:`pattern t`:sub:`1`:n:`, ..., t`:sub:`m` generates the
    equivalent goal
-   :g:`(fun (x`:sub:`1`:g:`:A`:sub:`1`:g:`) ... (x`:sub:`m` :g:`:A`:sub:`m` :g:`) =>`:math:`\varphi`:g:`(x`:sub:`1` :g:`... x`:sub:`m` :g:`)) t`:sub:`1` :g:`... t`:sub:`m`.
+   :g:`(fun (x`:sub:`1`:g:`:A`:sub:`1`:g:`) ... (x`:sub:`m` :g:`:A`:sub:`m` :g:`) =>
+   `:math:`\varphi`:g:`(x`:sub:`1` :g:`... x`:sub:`m` :g:`)) t`:sub:`1` :g:`... t`:sub:`m`.
```

Comment:
It would be much easier to write, with no risk of bad rendering (as is the case now) if we weren't using any `:math:` environment here and only the `:n:` quotation environment:

```
Starting from a goal :n:`T(t__1, ..., t__n)`, :tacn:`pattern` :n:`t__1, ..., t__n` generates...